pygame_gui/elements/ui_form.py

```diff
@@ -0,0 +1,1144 @@
+                # long_text (UITextEntryBox)
+                # integer (UITextEntryLine with only numeric inputs allowed)
+                # decimal (UITextEntryLine with numeric inputs and decimal point allowed)
+                # boolean (UIDropDownMenu with True and False as the options. Default option is True)
+
+                type_name, args = UIForm.type_checker(value)
+
+                # Two elements are created, a label and the actual UIElement needed for input
+
+                param_dict["relative_rect"].width = -1
+                param_dict["relative_rect"].height = label_height
+
+                # This is to prevent labels from resizing and becoming too small to hold their text
+                right = param_dict["anchors"].pop("right", None)
+                right_target = param_dict["anchors"].pop("right_target", None)
+
+                label = UILabel(text=key, object_id=ObjectID(f"#{key.lower()}", "@form_label"), **param_dict)
+
+                param_dict["anchors"]["top_target"] = label
+                if right:
+                    param_dict["anchors"]["right"] = right
+                if right_target:
+                    param_dict["anchors"]["right_target"] = right_target
+                param_dict["relative_rect"] = rel_rect.copy()
+                param_dict["relative_rect"].y = label_gap
+                param_dict["relative_rect"].height = field_height
+                param_dict["object_id"] = ObjectID(f"#{key.lower()}", "@form_field")
+
+                element = None
+
+                if type_name == "character":
+                    element = UITextEntryLine(initial_text=args.get("default", ""), **param_dict)
+                    element.set_text_length_limit(1)
+
+                elif type_name == "short_text":
+                    element = UITextEntryLine(initial_text=args.get("default", ""), **param_dict)
+
+                elif type_name == "long_text":
+                    param_dict["relative_rect"].height = large_field_height
+                    element = UITextEntryBox(initial_text=args.get("default", ""), **param_dict)
+
+                elif type_name == "password":
+                    element = UITextEntryLine(initial_text=args.get("default", ""), **param_dict)
+                    element.set_text_hidden(True)
+
+                elif type_name == "integer":
+                    element = UITextEntryLine(initial_text=args.get("default", ""), **param_dict)
+                    element.set_allowed_characters("numbers")
+
+                elif type_name == "decimal":
+                    element = UITextEntryLine(initial_text=args.get("default", ""), **param_dict)
+                    element.set_allowed_characters([".", *list("0123456789")])
+
+                elif type_name == "boolean":
+                    element = UIDropDownMenu(options_list=["True", "False"],
+                                             starting_option=args.get("default", "True"),
```

Comment:
```suggestion
                                             starting_option=str(args.get("default", "True")),
```

I had to add a cast here to get this code to run.

Reply:
When you create a boolean with a default argument, you're supposed to do it like `boolean(default="True")` and not without quotes as in `boolean(default=True)`, due to UIDropDown not accepting values except strings. Infact I thought that was already enforced. But now I change my mind, and think that it makes more sense if we do the casting behind the scenes ourselves, and then when we need to get the values, re-cast it to boolean as UIDropDown will only return strings as it's current option.